drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -57,6 +59,132 @@ def self.validate_json(json)
     recur_check.call(json)
   end
 
+  def hud_assessment?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) != 99
+  end
+
+  def intake?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) == 1
+  end
+
+  def exit?
+    Types::HmisSchema::Enums::AssessmentRole.as_data_collection_stage(role) == 3
+  end
+
+  def assessment_date_item
+    @assessment_date_item ||= link_id_item_hash.values.find(&:assessment_date)
+  end
+
+  def find_and_validate_assessment_date(hud_values:, entry_date:, exit_date:)
+    errors = HmisErrors::CustomValidationErrors.new
+    date = nil
+    item = assessment_date_item
+    readable_attribute = item.brief_text || item.text
+
+    if item.present? && hud_values.present?
+      date_string = hud_values[item.link_id]
+
+      if date_string.present?
+        date = HmisUtil::Dates.safe_parse_date(date_string: date_string, reasonable_years_distance: 10)
+        errors.add item.field_name, :invalid, readable_attribute: readable_attribute unless date.present?
+      else
+        errors.add item.field_name, :required, readable_attribute: readable_attribute
+      end
+    elsif hud_assessment?
+      errors.add :assessmentDate, :required, readable_attribute: readable_attribute
+    end
+
+    return [nil, errors.errors] if errors.errors.any?
+
+    # Additional validations for HUD assessment dates to be within appropriate entry/exit bounds
+    if date.present? && hud_assessment?
+      # Ensure assessment date is on or after entry date
+      errors.add item.field_name, :out_of_range, readable_attribute: readable_attribute, message: "must be after entry date (#{entry_date.strftime('%m/%d/%Y')})" if entry_date.present? && !intake? && date < entry_date
+      # Ensure assessment date is on or before exit date
+      errors.add item.field_name, :out_of_range, readable_attribute: readable_attribute, message: "must be before exit date (#{exit_date.strftime('%m/%d/%Y')})" if exit_date.present? && !exit? && date > exit_date
+    end
+
+    date = nil if errors.errors.any?
+
+    [date, errors.errors]
+  end
+
+  def validate_form_values(hud_values, _custom_values = nil)
+    errors = HmisErrors::CustomValidationErrors.new
+    hud_values.each do |link_id, value|
+      item = link_id_item_hash[link_id.to_s]
+      raise "Unrecognized link ID: #{link_id}" unless item.present?
+
+      next if item.assessment_date # Skip assessment date, it is validated separately
+
+      # Use item text as the "readable attribute" name to display on errors
+      readable_attribute = item.brief_text || item.text
+
+      is_missing = value.nil?
```

Comment:
we _might_ want `value.blank?`  Not sure if we ever have string values that could come through as `''`.  Maybe that's handled elsewhere.

Reply:
We shouldn't have empty strings come through, but I can update it to blank just in case